src/mapwindow.jl

```diff
@@ -165,47 +197,64 @@ function _indices_of_interiour_indices(fullimginds, imginds, kerinds)
     map(_indices_of_interiour_range, fullimginds, imginds, kerinds)
 end
 
-# replace median by ... outside of _mapwindow_kernel
-function _mapwindow_kernel(f,
+function allocate_output(f, img, window, border, imginds)
+    T = compute_output_eltype(f, img, window, border, imginds)
+    outinds = compute_output_indices(imginds)
+    similar(img, T, outinds)
+end
+
+function allocate_buffer(f, img, window)
+    T = eltype(img)
+    buf = Array{T}(map(length, window))
+    bufrs = default_shape(f)(buf)
+    buf, bufrs
+end
+
+function compute_output_eltype(f, img, window, border, imginds)
+    buf, bufrs = allocate_buffer(f, img, window)
+    make_buffer_values_realistic!(buf, img, window, border, imginds)
+    typeof(f(bufrs))
+end
+
+function make_buffer_values_realistic!(buf, img, window, border::Inner, imginds)
+    buf
```

Comment:
Yes. A hacky solution would be to fill the buffer with `one(eltype(img))`. A better solution would be to fill the buffer with data from the image. Doing that is a bit convoluted, since we need to make sure, that we stay in the inner of the image.

Reply:
Are you okay with filling the buffer with `one(eltype(img))`?

Reply:
Change that to `oneunit` and I think you have a deal. I only know of one case where this would fail:
```julia
julia> using FixedPointNumbers

julia> oneunit(Q0f7)
ERROR: InexactError()
Stacktrace:
 [1] convert at /home/tim/.julia/v0.6/FixedPointNumbers/src/fixed.jl:42 [inlined]
 [2] one at ./number.jl:163 [inlined]
 [3] oneunit(::Type{FixedPointNumbers.Fixed{Int8,7}}) at ./number.jl:186

julia> typemax(Q0f7)
0.992Q0f7
```